x-pack/platform/plugins/shared/triggers_actions_ui/public/application/components/rules_setting/rules_settings_flyout.test.tsx

```diff
@@ -0,0 +1,454 @@
+  statusChangeThreshold: 10,
+  createdBy: 'test user',
+  updatedBy: 'test user',
+  createdAt: new Date().toISOString(),
+  updatedAt: new Date().toISOString(),
+};
+const mockQueryDelaySetting: RulesSettingsQueryDelay = {
+  delay: 10,
+  createdBy: 'test user',
+  updatedBy: 'test user',
+  createdAt: new Date().toISOString(),
+  updatedAt: new Date().toISOString(),
+};
+
+const flyoutProps: RulesSettingsFlyoutProps = {
+  isVisible: true,
+  setUpdatingRulesSettings: jest.fn(),
+  onClose: jest.fn(),
+  onSave: jest.fn(),
+};
+
+const RulesSettingsFlyoutWithProviders: React.FunctionComponent<RulesSettingsFlyoutProps> = (
+  props
+) => (
+  <IntlProvider locale="en">
+    <QueryClientProvider client={queryClient}>
+      <RulesSettingsFlyout {...props} />
+    </QueryClientProvider>
+  </IntlProvider>
+);
+
+const waitForFlyoutLoad = async (options?: {
+  flappingSection?: boolean;
+  queryDelaySection?: boolean;
+}) => {
+  await waitFor(() => {
+    expect(screen.queryByTestId('rulesSettingsFlyout')).not.toBe(null);
+  });
+
+  const { flappingSection = true, queryDelaySection = true } = options || {};
+
+  await waitFor(() => {
+    if (flappingSection) {
+      expect(screen.queryByTestId('rulesSettingsFlappingSection')).toBeInTheDocument();
+    }
+    if (queryDelaySection) {
+      expect(screen.queryByTestId('rulesSettingsQueryDelaySection')).toBeInTheDocument();
+    }
+  });
+};
+
+describe('rules_settings_flyout', () => {
+  beforeEach(async () => {
+    jest.clearAllMocks();
+
+    const [
+      {
+        application: { capabilities },
+      },
+    ] = await mocks.getStartServices();
+    useKibanaMock().services.application.capabilities = {
+      ...capabilities,
+      rulesSettings: {
+        save: true,
+        show: true,
+        writeFlappingSettingsUI: true,
+        readFlappingSettingsUI: true,
+        writeQueryDelaySettingsUI: true,
+        readQueryDelaySettingsUI: true,
+      },
+    };
+
+    useKibanaMock().services.notifications.toasts = {
+      addSuccess: jest.fn(),
+      addError: jest.fn(),
+      addDanger: jest.fn(),
+      addWarning: jest.fn(),
+    } as unknown as IToasts;
+
+    useKibanaMock().services.isServerless = true;
+
+    fetchFlappingSettingsMock.mockResolvedValue(mockFlappingSetting);
+    updateFlappingSettingsMock.mockResolvedValue(mockFlappingSetting);
+    getQueryDelaySettingsMock.mockResolvedValue(mockQueryDelaySetting);
+    updateQueryDelaySettingsMock.mockResolvedValue(mockQueryDelaySetting);
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+    queryClient.clear();
+    cleanup();
+  });
+
+  test('renders flapping settings correctly', async () => {
+    const result = render(<RulesSettingsFlyoutWithProviders {...flyoutProps} />);
+    expect(fetchFlappingSettingsMock).toHaveBeenCalledTimes(1);
+    await waitForFlyoutLoad();
+    expect(
+      result.getByTestId('rulesSettingsFlappingEnableSwitch').getAttribute('aria-checked')
+    ).toBe('true');
```

Comment:
This is all a consequence of debugging too many cases' flaky tests ™️ but I got used to `await screen.findBy*` everywhere(vs `getBy*`).

I think it's fine here anyway, but I was wondering the following: 

```suggestion
    expect(
      (await screen.findByTestId('rulesSettingsFlappingEnableSwitch')).getAttribute('aria-checked')
    ).toBe('true');
```

would this work/be simpler?